src/form/Calendar/CalendarDays/CalendarDays.tsx

```diff
@@ -131,12 +171,36 @@ export const CalendarDays: FC<CalendarDaysProps> = ({
         Array.isArray(current) &&
         isSameDay(addDays(nextDayRangeEnd, -1), day.date);
 
+      // Determine styling of range start and end dates
```

Comment:
this is a little messy, but trying to determine whether or not to show the rounded corners wasn't as straightforward as you'd think